app/frontend/src/lib/currentLocation.js

```diff
@@ -0,0 +1,95 @@
+import '../polyfill/classlist.polyfill';
+import '../polyfill/after.polyfill';
+import '../polyfill/remove.polyfill';
+import '../loader';
+import { getPostcodeFromCoordinates } from './api';
+import { enableRadiusSelect, disableRadiusSelect } from '../search/ui/input/radius';
+
+const loader = new GOVUK.Loader();
+
+export const startLoading = (container, input) => {
+  input.disabled = true;
+  container.classList.add('js-location-finder--loading');
+  loader.init({
+    container: 'js-location-finder__input-container',
+    size: 32,
+    label: true,
+    labelText: 'Finding location...',
+  });
+};
+
+export const stopLoading = (container, input) => {
+  loader.stop();
+  container.classList.remove('js-location-finder--loading');
+  input.removeAttribute('disabled');
+};
+
+export const showLocationLink = (container) => {
+  container.classList.add('js-geolocation-supported');
+};
+
+export const showErrorMessage = (link) => {
+  if (!document.querySelector('.js-location-finder__link .govuk-error-message')) {
+    const errorMessage = document.createElement('div');
+    errorMessage.classList.add('govuk-error-message');
+    errorMessage.classList.add('govuk-!-margin-top-2');
+    errorMessage.innerHTML = 'Unable to find your location';
+    link.after(errorMessage);
+  }
+};
+
+export const removeErrorMessage = () => {
+  if (document.querySelector('.js-location-finder__link .govuk-error-message')) {
+    document.querySelector('.js-location-finder__link .govuk-error-message').remove();
+  }
+};
+
+export const onSuccess = (postcode, element) => {
+  element.value = postcode;
+  enableRadiusSelect();
+  currentLocation.stopLoading(document.querySelector('.js-location-finder'), document.querySelector('.js-location-finder__input'));
+};
+
+export const onFailiure = () => {
+  currentLocation.showErrorMessage(document.getElementById('current-location'));
+  disableRadiusSelect();
+  currentLocation.stopLoading(document.querySelector('.js-location-finder'), document.querySelector('.js-location-finder__input'));
+};
+
+export const postcodeFromPosition = (position) => getPostcodeFromCoordinates(position.coords.latitude, position.coords.longitude).then((response) => {
+  if (response.result) {
```

Comment:
Presuming `response.result` is always an array with at least one element if it's there, is it worth making sure?

Reply:
ill get this into the aforementioned test